cmd/ketch/pool_remove.go

```diff
@@ -35,14 +43,84 @@ type poolRemoveOptions struct {
 }
 
 func poolRemove(ctx context.Context, cfg config, options poolRemoveOptions, out io.Writer) error {
-	pool := ketchv1.Pool{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: options.Name,
-		},
+	var pool ketchv1.Pool
+
+	if err := cfg.Client().Get(ctx, types.NamespacedName{Name: options.Name}, &pool); err != nil {
+		return fmt.Errorf("failed to get pool: %w", err)
 	}
+
+	if userWantsToRemoveNamespace(pool.Spec.NamespaceName, out) {
+		if err := namespaceHasAdditionalResources(ctx, cfg, &pool); err != nil {
+			printNsRemovalErr(out, err)
+		} else {
+			if err := removeNamespace(ctx, cfg, &pool); err != nil {
+				printNsRemovalErr(out, err)
+			} else {
+				fmt.Fprintln(out, "Namespace successfully removed!")
+			}
+		}
+	}
+
 	if err := cfg.Client().Delete(ctx, &pool); err != nil {
 		return fmt.Errorf("failed to remove the pool: %w", err)
 	}
-	fmt.Fprintln(out, "Successfully removed!")
+
+	fmt.Fprintln(out, "Pool successfully removed!")
+
+	return nil
+}
+
+func userWantsToRemoveNamespace(ns string, out io.Writer) bool {
+	response := promptToRemoveNamespace(ns, out)
+	return handleNamespaceRemovalResponse(response, ns, out)
+}
+
+func promptToRemoveNamespace(ns string, out io.Writer) string {
+	fmt.Fprintf(out, "Do you want to remove the namespace along with the pool? Please enter namespace to confirm (%s): ", ns)
+
+	var response string
+	fmt.Scanln(&response)
+
+	return response
+}
+
+func handleNamespaceRemovalResponse(response, ns string, out io.Writer) bool {
+	if response != ns {
+		fmt.Fprintln(out, skipNsRemovalMsg)
+		return false
+	}
+
+	return true
+}
+
+func namespaceHasAdditionalResources(ctx context.Context, cfg config, targetPool *ketchv1.Pool) error {
+	var pools ketchv1.PoolList
+
+	if err := cfg.Client().List(ctx, &pools); err != nil {
+		return fmt.Errorf("failed to list pools: %w", err)
+	}
+
+	for _, p := range pools.Items {
+		if p.Name != targetPool.Name && p.Spec.NamespaceName == targetPool.Spec.NamespaceName {
+			return fmt.Errorf(
+				"Namespace contains other pools than %s, and cannot be removed:\nPools in target namespace:%+v",
+				targetPool.Name, pools.Items)
+		}
+	}
+
+	return nil
+}
+
+func removeNamespace(ctx context.Context, cfg config, pool *ketchv1.Pool) error {
+	var ns corev1.Namespace
+
+	if err := cfg.Client().Get(ctx, types.NamespacedName{Name: pool.Spec.NamespaceName}, &ns); err != nil {
+		return fmt.Errorf("failed to get namespace: %s", err)
```

Comment:
You should wrap the returned error with the %w verb instead of %s

Reply:
👍